packages/devtools_app/lib/src/trees.dart

```diff
@@ -138,18 +142,73 @@ class TreeNode<T extends TreeNode<T>> {
     );
   }
 
-  T firstNodeAtLevel(int level) {
+  /// Locates the first sub-node in the tree at level [level].
+  ///
+  /// [level] is relative to the subtree root [this].
+  ///
+  /// For example:
+  ///
+  /// [level 0]                A
+  ///                        /   \
+  /// [level 1]             B     E
+  ///                      /    /  \
+  /// [level 2]           C    F    G
+  ///                    /
+  /// [level 3]         D
+  ///
+  /// E.firstSubNodeAtLevel(1) => F
+  T firstSubNodeAtLevel(int level) {
+    return _subNodeAtLevelWithCondition(
+      level,
+      // When this condition is called, we have already ensured that
+      // [level] < [depth], so at least one child is guaranteed to meet the
+      // firstWhere condition.
+      (currentNode, levelWithOffset) => currentNode.children
+          .firstWhere((n) => n.depth + n.level > levelWithOffset),
+    );
+  }
+
+  /// Locates the last sub-node in the tree at level [level].
+  ///
+  /// [level] is relative to the subtree root [this].
+  ///
+  /// For example:
+  ///
+  /// [level 0]                A
+  ///                        /   \
+  /// [level 1]             B     E
+  ///                      /    /  \
+  /// [level 2]           C    F    G
+  ///                    /
+  /// [level 3]         D
+  ///
+  /// E.lastSubNodeAtLevel(1) => G
+  T lastSubNodeAtLevel(int level) {
```

Comment:
same. remove the word "sub". I'm open to adding child but "sub" seems like a bit non-standard terminology unless I'm missing something.